### PR TITLE
[service] add internal factory method for meter provider

### PR DESCRIPTION
**Pull request by codeboten**
This will be used when moving the meter provider configuration to use the otel-go config package.

Part of #10414

Skipping changelog as there's no usage of this method yet. It will be done in a separate PR

### CODE REVIEW

**Comment by codeboten** [comment]
Ping @mx-psi for review

File: service/telemetry/internal/factory.go [hunk 1]
```diff
@@ -102,6 +106,21 @@ func (f *factory) CreateTracerProvider(ctx context.Context, set CreateSettings,
 	return f.CreateTracerProviderFunc(ctx, set, cfg)
 }
 
+type CreateMeterProviderFunc func(context.Context, CreateSettings, component.Config) (metric.MeterProvider, error)
+
+func WithMeterProvier(createMeterProvider CreateMeterProviderFunc) FactoryOption {
```
Comment (mx-psi) [inline service/telemetry/internal/factory.go:111]:
```suggestion
// WithMeterProvider overrides the default no-op meter provider.
func WithMeterProvider(createMeterProvider CreateMeterProviderFunc) FactoryOption {
```

File: service/telemetry/internal/factory.go [hunk 2]
```diff
@@ -102,6 +106,21 @@ func (f *factory) CreateTracerProvider(ctx context.Context, set CreateSettings,
 	return f.CreateTracerProviderFunc(ctx, set, cfg)
 }
 
+type CreateMeterProviderFunc func(context.Context, CreateSettings, component.Config) (metric.MeterProvider, error)
```
Comment (mx-psi) [inline service/telemetry/internal/factory.go:109]:
```suggestion
// CreateMeterProviderFunc is the equivalent of Factory.CreateMeterProvider.
type CreateMeterProviderFunc func(context.Context, CreateSettings, component.Config) (metric.MeterProvider, error)
```

**Review by mx-psi — APPROVED** [review]

File: service/telemetry/internal/factory.go [hunk 3]
```diff
@@ -34,7 +36,8 @@ type Factory interface {
 	// CreateTracerProvider creates a TracerProvider.
 	CreateTracerProvider(ctx context.Context, set Settings, cfg component.Config) (trace.TracerProvider, error)
 
-	// TODO: Add CreateMeterProvider.
+	// CreateMeterProvider creates a MeterProvider.
+	CreateMeterProvider(ctx context.Context, set CreateSettings, cfg component.Config) (metric.MeterProvider, error)
```
Comment (codeboten) [inline service/telemetry/internal/factory.go:40]:
```suggestion
	CreateMeterProvider(ctx context.Context, set Settings, cfg component.Config) (metric.MeterProvider, error)
```

File: service/telemetry/internal/factory.go [hunk 4]
```diff
@@ -102,6 +106,23 @@ func (f *factory) CreateTracerProvider(ctx context.Context, set Settings, cfg co
 	return f.CreateTracerProviderFunc(ctx, set, cfg)
 }
 
+// CreateMeterProviderFunc is the equivalent of Factory.CreateMeterProvider.
+type CreateMeterProviderFunc func(context.Context, CreateSettings, component.Config) (metric.MeterProvider, error)
```
Comment (codeboten) [inline service/telemetry/internal/factory.go:110]:
```suggestion
type CreateMeterProviderFunc func(context.Context, Settings, component.Config) (metric.MeterProvider, error)
```

File: service/telemetry/internal/factory.go [hunk 5]
```diff
@@ -102,6 +106,23 @@ func (f *factory) CreateTracerProvider(ctx context.Context, set Settings, cfg co
...
+
+// WithMeterProvider overrides the default no-op meter provider.
+func WithMeterProvider(createMeterProvider CreateMeterProviderFunc) FactoryOption {
+	return factoryOptionFunc(func(o *factory) {
+		o.CreateMeterProviderFunc = createMeterProvider
+	})
+}
+
+func (f *factory) CreateMeterProvider(ctx context.Context, set CreateSettings, cfg component.Config) (metric.MeterProvider, error) {
```
Comment (codeboten) [inline service/telemetry/internal/factory.go:119]:
```suggestion
func (f *factory) CreateMeterProvider(ctx context.Context, set Settings, cfg component.Config) (metric.MeterProvider, error) {
```

**Pull request merged**